pkg/protocol/xprotocol/dubbo/deconder.go

```diff
@@ -0,0 +1,128 @@
+package dubbo
+
+import (
+	"context"
+	"encoding/binary"
+	"fmt"
+
+	"github.com/AlexStocks/dubbogo/codec/hessian"
+	"mosn.io/mosn/pkg/buffer"
+	"mosn.io/mosn/pkg/types"
+)
+
+func decodeFrame(ctx context.Context, data types.IoBuffer) (cmd interface{}, err error) {
+	// convert data to duboo frame
+	dataBytes := data.Bytes()
+	frame := &Frame{
+		rawData: dataBytes,
+		data:    data,
+	}
+	// decode magic
+	frame.Magic = dataBytes[MagicIdx:FlagIdx]
+	// decode flag
+	frame.Flag = dataBytes[FlagIdx]
+	// decode status
+	frame.Status = dataBytes[StatusIdx]
+	// decode request id
+	reqIdRaw := dataBytes[IdIdx:(IdIdx + IdLen)]
+	frame.Id = binary.BigEndian.Uint64(reqIdRaw)
+	// decode data length
+	frame.DataLen = binary.BigEndian.Uint32(dataBytes[DataLenIdx:(DataLenIdx + DataLenSize)])
+
```

Comment:
Previous check only ensures that buffer size is  larger than HeaderLen, but no perform check for (HeaderLen + DataLen)

Reply:
fixed